### PR TITLE
1356 Reintroduce Frida Stalker

### PR DESCRIPTION
**Pull request by cpholguera**
- Introduced proper description for Stalker and Interceptor.
- The main differences were highlighted.
- Enhanced Frida intro in 6b.
- Added Links to Frida from 5b to 4c.

This PR covers issue #1356.

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit 1abaa827da4758406bca5e5c6512bbcb5a4bbb64: Broken link result: [!] Error(s) found by the Links validator: 6 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       12 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 7be8dbdefae239ae17cbb96918fab4ac45ac52df: Broken link result: [!] Error(s) found by the Links validator: 7 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       12 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x04c-Tampering-and-Reverse-Engineering.md [hunk 1]
```diff
@@ -74,6 +74,13 @@ Frida offers three modes of operation:
 2. Embedded: this is the case when your device is rooted or jailbroken (you cannot use ptrace as an unprivileged user), you're responsible for the injection of the [frida-gadget](https://www.frida.re/docs/gadget/ "Frida Gadget") library by embedding it into your app.
```
[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x04c-Tampering-and-Reverse-Engineering.md]: Thanks Sven! In fact, that's actually a "bug". The text should read "Embedded: this is the case when your device is not rooted or jailbroken"

I'll correct that!

Comment (cpholguera) [inline Document/0x04c-Tampering-and-Reverse-Engineering.md]:
```suggestion
2. Embedded: this is the case when your device is not rooted or jailbroken (you cannot use ptrace as an unprivileged user), you're responsible for the injection of the [frida-gadget](https://www.frida.re/docs/gadget/ "Frida Gadget") library by embedding it into your app.
```

[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x04c-Tampering-and-Reverse-Engineering.md]: Done :)

**Comment by OMTGreleaser** [comment]
Results for commit 2f1d47bfd4e26b8b635ab62b70612e39a34960ee: Broken link result: [!] Error(s) found by the Links validator: 7 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       12 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x04c-Tampering-and-Reverse-Engineering.md [hunk 2]
```diff
@@ -71,9 +71,16 @@ In contrast, Frida implements code injection by writing code directly into proce
...
 3. Preloaded: similar to `LD_PRELOAD` or `DYLD_INSERT_LIBRARIES`. You can configure the frida-gadget to run autonomously and load a script from the filesystem (e.g. path relative to where the Gadget binary resides).
 
+Independently of the chosen mode, you can make use of the [Frida JavaScript APIs](https://www.frida.re/docs/javascript-api/ "Frida JavaScript APIs") to interact with the running process and its memory. Some of the fundamental APIs are:
+
+- [Interceptor](https://www.frida.re/docs/javascript-api/#interceptor "Interceptor"): When using the Interceptor API, Frida injects a trampoline (aka in-line hooking) at the function prologue which provokes a redirection to our custom code, executes our code, and returns to the original function. Note that while very effective for our purpose, this introduces a considerable overhead (due to the trampoline related jumping and context switching) and cannot be considered transparent as it overwrites the original code and acts similar to a debugger (putting breakpoints) and therefore can be detected on a similar manner, e.g. by applications that periodically checksum their own code.
```
Comment (TheDauntless) [inline Document/0x04c-Tampering-and-Reverse-Engineering.md]:
```suggestion
- [Interceptor](https://www.frida.re/docs/javascript-api/#interceptor "Interceptor"): When using the Interceptor API, Frida injects a trampoline (aka in-line hooking) at the function prologue which provokes a redirection to our custom code, executes our code, and returns to the original function. Note that while very effective for our purpose, this introduces a considerable overhead (due to the trampoline related jumping and context switching) and cannot be considered transparent as it overwrites the original code and acts similar to a debugger (putting breakpoints) and therefore can be detected in a similar manner, e.g. by applications that periodically checksum their own code.
```

File: Document/0x06b-Basic-Security-Testing.md [hunk 3]
```diff
@@ -172,7 +172,27 @@ In order to analyze iOS apps, you should install the following tools on your hos
...
+[Frida](https://www.frida.re "Frida") is a free and open-source dynamic code instrumentation toolkit that lets you execute snippets of JavaScript into your native apps. It was already introduced in the chapter "[Tampering and Reverse Engineering](0x04c-Tampering-and-Reverse-Engineering.md#frida "Frida")" of the general testing guide. Frida is used in several of the following sections and chapters.
+
+Frida supports interaction with the Objective-C runtime though the [ObjC API](https://www.frida.re/docs/javascript-api/#objc "Frida - ObjC API"). You'll be able to hook and call both Objective-C and native functions inside the process and its native libraries. Your JavaScript snippets have full access to memory, e.g. to read and/or write any structured data.
```
Comment (TheDauntless) [inline Document/0x06b-Basic-Security-Testing.md]:
```suggestion
Frida supports interaction with the Objective-C runtime through the [ObjC API](https://www.frida.re/docs/javascript-api/#objc "Frida - ObjC API"). You'll be able to hook and call both Objective-C and native functions inside the process and its native libraries. Your JavaScript snippets have full access to memory, e.g. to read and/or write any structured data.
```

File: Document/0x06b-Basic-Security-Testing.md [hunk 4]
```diff
@@ -172,7 +172,27 @@ In order to analyze iOS apps, you should install the following tools on your hos
...
+
+Here are some tasks that Frida APIs offers and are relevant or exclusive on iOS:
+
+- Instantiate Objective-C objects and call static and non-static class methods ([ObjC API](https://www.frida.re/docs/javascript-api/#objc "Frida - ObjC API")).
+- Trace Objective-C method calls and/or replace their implementations ([Interceptor API](https://www.frida.re/docs/javascript-api/#interceptor "Frida - Interceptor API")).
+- Enumerate live instances of specific classes by scanning the heap ([ObjC API](https://www.frida.re/docs/javascript-api/#objc "Frida - ObjC API")).
+- Scan process memory for occurrences of a string ([Memory API](https://www.frida.re/docs/javascript-api/#memory "Frida - Memory API")).
+- Intercept native function calls to run your own code at function entry and exit ([Interceptor API](https://www.frida.re/docs/javascript-api/#interceptor "Frida - Interceptor API")).
+
+Remember that on iOS, you can also benefit from the built-in tools provided when installing Frida, that includes the Frida CLI (`frida`), `frida-ps`, `frida-ls-devices` and `frida-trace`, to name some of them.
```
Comment (TheDauntless) [inline Document/0x06b-Basic-Security-Testing.md]:
```suggestion
Remember that on iOS, you can also benefit from the built-in tools provided when installing Frida, which include the Frida CLI (`frida`), `frida-ps`, `frida-ls-devices` and `frida-trace`, to name a few.
```

File: Document/0x06b-Basic-Security-Testing.md [hunk 5]
```diff
@@ -172,7 +172,27 @@ In order to analyze iOS apps, you should install the following tools on your hos
...
+
+There's a `frida-trace` feature exclusive on iOS worth highlighting: tracing Objective-C APIs using the `-m` flag and wildcards. For example, tracing all methods including "HTTP" in its name and belonging to any class whose name starts with "NSURL" is as easy as running:
```
Comment (TheDauntless) [inline Document/0x06b-Basic-Security-Testing.md]:
```suggestion
There's a `frida-trace` feature exclusive on iOS worth highlighting: tracing Objective-C APIs using the `-m` flag and wildcards. For example, tracing all methods including "HTTP" in their name and belonging to any class whose name starts with "NSURL" is as easy as running:
```

File: Document/0x04c-Tampering-and-Reverse-Engineering.md [hunk 6]
```diff
@@ -54,7 +54,7 @@ We'll include examples of all three frameworks. We recommend starting with Frida
 
 ##### Frida
 
-[Frida](https://www.frida.re "Frida") is a free and open source dynamic code instrumentation toolkit written in C that works by injecting a JavaScript engine ([Duktape](https://duktape.org/ "Duktape JavaScript Engine") and [V8](https://v8.dev/docs "V8 JavaScript Engine")) into the instrumented process. Frida lets you execute snippets of JavaScript into native apps on Android and iOS (as well as on [other platforms](https://www.frida.re/docs/home/ "So what is Frida, exactly?")).
+[Frida](https://www.frida.re "Frida") is a free and open source dynamic code instrumentation toolkit written in C by Ole André Vadla Ravnås that works by injecting a JavaScript engine ([Duktape](https://duktape.org/ "Duktape JavaScript Engine") and [V8](https://v8.dev/docs "V8 JavaScript Engine")) into the instrumented process. Frida lets you execute snippets of JavaScript into native apps on Android and iOS (as well as on [other platforms](https://www.frida.re/docs/home/ "So what is Frida, exactly?")).
```
Comment (cpholguera) [inline Document/0x04c-Tampering-and-Reverse-Engineering.md]:
```suggestion
[Frida](https://www.frida.re "Frida") is a free and open source dynamic code instrumentation toolkit written in Vala by Ole André Vadla Ravnås that works by injecting a JavaScript engine ([Duktape](https://duktape.org/ "Duktape JavaScript Engine") and [V8](https://v8.dev/docs "V8 JavaScript Engine")) into the instrumented process. Frida lets you execute snippets of JavaScript into native apps on Android and iOS (as well as on [other platforms](https://www.frida.re/docs/home/ "So what is Frida, exactly?")).
```

[the remark this answers is not in the document]
Comment (cpholguera) [reply Document/0x04c-Tampering-and-Reverse-Engineering.md]: In deed, nice catch ;)

**Comment by OMTGreleaser** [comment]
Results for commit 65bf1a4d2f2e0cbab54a9849f890774bb23aa2fa: Broken link result: [!] Error(s) found by the Links validator: 6 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       13 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 9de259ef6863f44cf7767fe13895d3ba78467bc6: Broken link result: [!] Error(s) found by the Links validator: 6 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       13 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Pull request merged**